sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala

```diff
@@ -137,7 +137,11 @@ trait WindowExecBase extends UnaryExecNode {
           function match {
             case AggregateExpression(f, _, _, _, _) => collect("AGGREGATE", frame, e, f)
             case f: AggregateWindowFunction => collect("AGGREGATE", frame, e, f)
-            case f: OffsetWindowFunction => collect("OFFSET", frame, e, f)
+            case f: OffsetWindowFunction => if (f.startWithCurrentRow) {
+              collect("OFFSET", frame, e, f)
+            } else {
+              collect("WHOLE_OFFSET", frame, e, f)
```

Comment:
we need a better name than `WHOLE_OFFSET`

Reply:
OK

Reply:
ok